File: src/web/src/app/model/customer.ts
export class Customer{
  customerId?: number;
  firstName?: string;
  lastName?: string;
  birthDate?: Date;

  constructor(
    customerId?: number,
    firstName?: string,
    lastName?: string,
    birthDate?: Date
  ) {
      this.customerId = customerId;
      this.firstName = firstName;
      this.lastName = lastName;
      this.birthDate = birthDate;
    }
}

File: src/main/java/allane/model/Customer.java
package allane.model;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Customer {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    @Column(name = "customer_id")
    private long customerId;
    private String firstName;
    private String lastName;

    @JsonSerialize(using = LocalDateSerializer.class)
    @JsonDeserialize(using = LocalDateDeserializer.class)
    private LocalDate birthDate;
}
